data_platform/utils/deduplication.py

```
@@ -0,0 +1,66 @@
+
+from dataclasses import dataclass
+from pathlib import Path
+from typing import TYPE_CHECKING, Any
+
+if TYPE_CHECKING:
+    from data_platform.utils.storage import StorageManager
+
+PRIOR_RUN_POLICIES = frozenset({"prior_runs_same_dataset", "prior_runs_all_datasets"})
+
+
+def policy_includes_prior_runs(policy: Any) -> bool:
```

Ruff (0.16.1)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `policy`

(ANN401)